doc/sql/semantics.rst

```diff
@@ -0,0 +1,239 @@
+Semantics of built-in SQL operations
+====================================
+
+Nullable types
+--------------
+
+A type is nullable if it can represent the ``NULL`` value.  For input
+tables the nullability of a column is declared explicitly.  For
+intermediate results and output views the compiler infers the
+nullability of each column using type inference rules.
+
+Most SQL operations are defined for nullable types.  Our compiler
+follows the SQL standard in this respect.  Most operations (e.g.,
+``+``), when applied a ``NULL`` operand will produce a ``NULL``
+value.
+
+Boolean Operations
+------------------
+
+We support the following Boolean operations: ``OR``, ``AND``, ``NOT``,
+``IS FALSE``, ``IS NOT FALSE``, ``IS TRUE``, ``IS NOT TRUE``.  Notice
+that not all Boolean operations produce ``NULL`` results when an
```

Comment:
Unable to 'notice' null behavior from the text. We should just write out which ones produce null when the operand is null.

Reply:
This should be fixed in #105